src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -1776,10 +1493,9 @@ FullWidthRepeat:
 
             ' // First read a leading base specifier, if present, followed by a sequence of zero
             ' // or more digits.
-            Dim ch = Peek()
-            If ch = "&"c OrElse ch = FULLWIDTH_AMPERSAND Then
+            If ch.IsAnyOf("&"c, FULLWIDTH_AMPERSAND) Then
                 Here += 1
-                ch = If(CanGet(Here), Peek(Here), ChrW(0))
+                If Not TryPeek(Here,ch) Then ch = ChrW(0)
```

Comment:
Consider using multi-line If. Alternatively TryPeek could always assign ChrW(0) to the output parameter on failure, then 'if' wouldn't be needed.